setup_libuv.py

```diff
@@ -1,169 +1,135 @@
+        'deps/libuv/src/unix/dl.c',
+        'deps/libuv/src/unix/fs.c',
+        'deps/libuv/src/unix/getaddrinfo.c',
+        'deps/libuv/src/unix/getnameinfo.c',
+        'deps/libuv/src/unix/loop-watcher.c',
+        'deps/libuv/src/unix/loop.c',
+        'deps/libuv/src/unix/pipe.c',
+        'deps/libuv/src/unix/poll.c',
+        'deps/libuv/src/unix/process.c',
+        'deps/libuv/src/unix/signal.c',
+        'deps/libuv/src/unix/stream.c',
+        'deps/libuv/src/unix/tcp.c',
+        'deps/libuv/src/unix/thread.c',
+        'deps/libuv/src/unix/timer.c',
+        'deps/libuv/src/unix/tty.c',
+        'deps/libuv/src/unix/udp.c',
+    ]
 
 
 class libuv_build_ext(build_ext):
-    libuv_dir      = os.path.join('deps', 'libuv')
-    libuv_repo     = 'https://github.com/libuv/libuv.git'
-    libuv_branch   = 'v1.x'
-    libuv_revision = 'd6ac141'
-    libuv_patches  = []
+    libuv_dir = os.path.join('deps', 'libuv')
 
     user_options = build_ext.user_options
     user_options.extend([
-        ("libuv-clean-compile", None, "Clean libuv tree before compilation"),
-        ("libuv-force-fetch", None, "Remove libuv (if present) and fetch it again"),
-        ("libuv-verbose-build", None, "Print output of libuv build process"),
-        ("use-system-libuv", None, "Use the system provided libuv, instead of the bundled one")
+        ("use-system-libuv", None, "Use the system provided libuv, instead of the bundled one"),
     ])
     boolean_options = build_ext.boolean_options
-    boolean_options.extend(["libuv-clean-compile", "libuv-force-fetch", "libuv-verbose-build", "use-system-libuv"])
+    boolean_options.extend(["use-system-libuv"])
 
     def initialize_options(self):
         build_ext.initialize_options(self)
-        self.libuv_clean_compile = 0
-        self.libuv_force_fetch = 0
-        self.libuv_verbose_build = 0
         self.use_system_libuv = 0
 
     def build_extensions(self):
-        self.force = self.force or self.libuv_force_fetch or self.libuv_clean_compile
-
+        sources = SOURCES
+        # XXX Is mingw32 check correct here?!
         if self.compiler.compiler_type == 'mingw32' or self.use_system_libuv:
             self.compiler.add_library('uv')
         else:
-            if sys.platform == 'win32':
-                self.libuv_lib = os.path.join(self.libuv_dir, 'Release', 'lib', 'libuv.lib')
-            else:
-                self.libuv_lib = os.path.join(self.libuv_dir, '.libs', 'libuv.a')
-            self.get_libuv()
-            # Set compiler options
-            self.extensions[0].extra_objects.extend([self.libuv_lib])
             self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'include'))
+            self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'src'))
+            if sys.platform != 'win32':
+                self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'src', 'unix'))
+
+            if sys.platform.startswith('linux'):
+                sources += [
```

Comment:
can we move the sources to the ifs at the toplevel? Then here we'd just deal with flags and the like.